File: CS/S37186_MVC/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using S37186_MVC.Helpers;

namespace Q522482_2.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";

            return View();
        }
        [ValidateInput(false)]
        public ActionResult FileManagerPartial()
        {
            return PartialView("_FileManagerPartial", HomeControllerFileManagerSettings.Model);
        }

        public FileStreamResult FileManagerPartialDownload()
        {
            return FileManagerExtension.DownloadFiles(FMSettingsHelper.CreateFileManagerDownloadSettings(), HomeControllerFileManagerSettings.Model);
        }
    }
    public class HomeControllerFileManagerSettings
    {
        public const string RootFolder = @"~\";

        public static string Model { get { return RootFolder; } }
    }


}

File: VB/S37186_MVC_VB3/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace S37186_MVC_VB3.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"
            Return View()
        End Function
        <ValidateInput(False)> _
        Public Function FileManagerPartial() As ActionResult
            Return PartialView("_FileManagerPartial", HomeControllerFileManagerSettings.Model)
        End Function

        Public Function FileManagerPartialDownload() As FileStreamResult
            Return FileManagerExtension.DownloadFiles("FileManager", HomeControllerFileManagerSettings.Model)
        End Function
    End Class
    Public Class HomeControllerFileManagerSettings
        Public Const RootFolder As String = "~\"

        Public Shared ReadOnly Property Model() As String
            Get
                Return RootFolder
            End Get
        End Property
    End Class


End Namespace
